File: testdata/javascript/simple.js
// Simple JavaScript functions for testing

function add(a, b) {
    return a + b;
}

const multiply = function(a, b) {
    return a * b;
};

const divide = (a, b) => a / b;

async function fetchData() {
    return 42;
}

class Calculator {
    constructor(initial = 0) {
        this.value = initial;
    }

    add(x) {
        this.value += x;
        return this.value;
    }

    getValue() {
        return this.value;
    }
}

File: testdata/c/simple.c
int multiply(int a, int b) {
    return a * b;
}

int divide(int a, int b) {
    if (b == 0) return 0;
    return a / b;
}

int factorial(int n) {
    if (n <= 1) return 1;
    return n * factorial(n - 1);
}
